### ISSUE TITLE
Fix onPlayerDeath trigger

### ISSUE THREAD

**Issue by ElPumpo** [ESX-Org/es_extended#118]
Will need to do futher debugging. If someone has a stack trace it could go a long way.

**Comment by SaltyGrandpa** [comment]
```
[    303031] Error running system event handling function for resource es_extended: citizen:/scripting/lua/scheduler.lua:41: Failed to execute thread: client/main.lua:103: attempt to index a number value (local 'victimCoords')
[    303031] stack traceback:
[    303031] 	client/main.lua:103: in upvalue 'handler'
[    303031] 	citizen:/scripting/lua/scheduler.lua:175: in function <citizen:/scripting/lua/scheduler.lua:174>
[    303031] stack traceback:
[    303031] 	[C]: in function 'error'
[    303031] 	citizen:/scripting/lua/scheduler.lua:41: in field 'CreateThreadNow'
[    303031] 	citizen:/scripting/lua/scheduler.lua:174: in function <citizen:/scripting/lua/scheduler.lua:138>
```

**Comment by ElPumpo** [comment]
Yep sometimes random variables are not found, we'll have to add extra checks.

**Comment by SaltyGrandpa** [comment]
Here is a dump of the `data` table when this occurs (before the extra metadata is inserted).

```
[    143843] {
[    143843] ["weaponhash"] = 453432689,
[    143843] ["killerpos"] = 105.00686645508,
[    143843] ["killervehname"] = ,
[    143843] ["killertype"] = 4,
[    143843] ["killervehseat"] = 0,
[    143843] ["killerinveh"] = false,
[    143843] }
[    143843] Error running system event handling function for resource es_extended: citizen:/scripting/lua/scheduler.lua:41: Failed to execute thread: client/main.lua:104: attempt to index a number value (local 'victimCoords')
[    143843] stack traceback:
[    143843] 	client/main.lua:104: in upvalue 'handler'
[    143859] 	citizen:/scripting/lua/scheduler.lua:175: in function <citizen:/scripting/lua/scheduler.lua:174>
[    143859] stack traceback:
[    143859] 	[C]: in function 'error'
[    143859] 	citizen:/scripting/lua/scheduler.lua:41: in field 'CreateThreadNow'
[    143859] 	citizen:/scripting/lua/scheduler.lua:174: in function <citizen:/scripting/lua/scheduler.lua:138>
```

Appears to only happen when NPCs kill a player. Or at least that's how I've been reproducing it. /die doesn't cause any issues. 

Let me know if I can help.

**Comment by ElPumpo** [comment]
Well a check if the player id is valid is what we need

**Comment by SaltyGrandpa** [comment]
Would `NetworkIsPlayerActive` suffice for this? If so, I'll find an elegant way to throw that in there. Unless you have another/better idea, don't want to step on toes.

**Comment by ElPumpo** [comment]
Exactly, I'll do further tests in the #future

**Comment by SimplyLulle** [comment]
Hi, I have a problem with the ambulance job so when I or someone else revives someone it will be completely black screen for the person we revive; As well as with / revive

Can somebody help? ; i think somebody can help 📦

**Comment by ElPumpo** [comment]
7e31fc90f4e589323a5a8d5d7035023186094774 should have fixed the issue. I'll have to run even more tests -_-

**Comment by SLuuuuuuG** [comment]
The problem still happens with the latest update

**Comment by ElPumpo** [comment]
Hi. Can you please send a stack trace?

**Comment by SLuuuuuuG** [comment]
http://oi65.tinypic.com/2zhl6if.jpg

**Comment by SLuuuuuuG** [comment]
This is the error I am receiving after @SaltyGrandpa 's edit 

http://oi65.tinypic.com/289lf2t.jpg

**Comment by ElPumpo** [comment]
Ok I've now ditched baseevents fully as it casued various issues in certian scenarios when getting killed by someone in an vehicle, and it also had compability issues with seat-belt scripts.

The code was also messy and had misleading variables. I ask you all to try this new version, it should work just fine now, but if it doesn't than post a stack trace and I'll take a further look since I haven't fully tested the 'killed by player' thing due to lack of time.

The `data` content has been changed again, with client and server ID of the killer. See the source code for the module, the ESX docs remain outdated til I get it updated..
